src/code42cli/cmds/legal_hold.py

```diff
@@ -241,37 +256,50 @@ def _get_and_print_preservation_policy(sdk, policy_uid):
     echo(pformat(json.loads(preservation_policy.text)))
 
 
-def _get_legal_hold_membership_id_for_user_and_matter(sdk, username, matter_id):
-    user_id = get_user_id(sdk, username)
-    memberships = _get_legal_hold_memberships_for_matter(sdk, matter_id, active=True)
+def _get_legal_hold_membership_id_for_user_and_matter(state, username, matter_id):
+    user_id = get_user_id(state.sdk, username)
+    memberships = _get_legal_hold_memberships_for_matter(state, matter_id, active=True)
     for member in memberships:
         if member["user"]["userUid"] == user_id:
             return member["legalHoldMembershipUid"]
     raise UserNotInLegalHoldError(username, matter_id)
 
 
-def _get_legal_hold_memberships_for_matter(sdk, matter_id, active=True):
-    memberships_generator = sdk.legalhold.get_all_matter_custodians(
-        legal_hold_uid=matter_id, active=active
+def _get_legal_hold_memberships_for_matter(state, matter_id, active=True):
+    memberships_generator = state.sdk.legalhold.get_all_matter_custodians(
+        matter_id, active=active
     )
-    memberships = [
-        member
-        for page in memberships_generator
-        for member in page["legalHoldMemberships"]
-    ]
+    if state.profile.api_client_auth == "True":
+        memberships = [member for page in memberships_generator for member in page]
+    else:
+        memberships = [
+            member
+            for page in memberships_generator
+            for member in page["legalHoldMemberships"]
+        ]
     return memberships
 
 
-def _get_all_active_matters(sdk):
-    matters_generator = sdk.legalhold.get_all_matters()
-    matters = [
-        matter
-        for page in matters_generator
-        for matter in page["legalHolds"]
-        if matter["active"]
-    ]
-    for matter in matters:
-        matter["creator_username"] = matter["creator"]["username"]
+def _get_all_active_matters(state):
+    matters_generator = state.sdk.legalhold.get_all_matters()
+    if state.profile.api_client_auth == "True":
+        matters = [
+            matter for page in matters_generator for matter in page if matter["active"]
+        ]
+        for matter in matters:
+            try:
+                matter["creator_username"] = matter["creator"]["user"]["email"]
```

Comment:
This will fail if the matter was created via an API client, right?

Reply:
I'm not sure what the conditions for the field being populated are (but that would make sense).  This `creator user email` field isn't even included on the response documentation and isn't on every matter, but it does exist and is populated on some of them.